test/integration/curator_test.rb

```diff
@@ -151,36 +201,104 @@ def test_herbarium_change_code
     assert_not_equal(new_code, herbarium.code)
     curator = herbarium.curators[0]
     login!(curator.login, "testpassword", true)
-    get("/herbarium/edit_herbarium?id=#{herbarium.id}")
-    open_form do |form|
-      form.assert_value("code", herbarium.code)
-      form.change("code", new_code)
+    get(edit_herbarium_path(herbarium))
+    open_form(
+      # The edit form posts to update; this is the update url
+      "form[action^='#{herbarium_path(herbarium)}']"
+    ) do |form|
+      form.assert_value("[code]", herbarium.code)
+      form.change("[code]", new_code)
       form.submit(:SAVE.t)
     end
-    herbarium = Herbarium.find(herbarium.id)
-    assert_equal(new_code, herbarium.code)
-    assert_template("herbarium/show_herbarium")
+
+    assert_equal(new_code, herbarium.reload.code)
+    assert_select(
+      "#title-caption",
+      { text: herbarium.format_name },
+      "Changing Fungarium code should land on page for that Fungarium"
+    )
   end
 
-  def test_herbarium_create
+  def test_herbarium_create_and_destroy
     user = users(:mary)
     assert_equal([], user.curated_herbaria)
     login!(user.login, "testpassword", true)
-    get("/herbarium/create_herbarium")
-    open_form do |form|
-      form.assert_value("herbarium_name", "")
-      form.assert_value("code", "")
-      form.assert_value("place_name", "")
-      form.assert_value("email", "")
-      form.assert_value("mailing_address", "")
-      form.assert_value("description", "")
-      form.assert_unchecked("personal")
-      form.change("herbarium_name", "Mary's Herbarium")
-      form.check("personal")
-      form.submit(:CREATE.t)
+    get(herbaria_alls_path)
+    click(label: :create_herbarium.l)
+
+    open_form("form[action^='#{herbaria_path}']") do |form|
+      form.assert_value("[name]", "")
+      form.assert_value("[code]", "")
+      form.assert_value("[place_name]", "")
+      form.assert_value("[email]", "")
+      form.assert_value("[mailing_address]", "")
+      form.assert_value("[description]", "")
+      form.assert_unchecked("[personal]")
+
+      form.change("[name]", "Mary's Herbarium")
+      form.check("[personal]")
+      form.submit(:CREATE.l)
```

Comment:
New style forms (using [Model Object Helpers](https://guides.rubyonrails.org/v5.2/form_helpers.html#dealing-with-model-objects-model-object-helpers) instead of `_tag` do not create `id`s. 
So I tweaked/hacked `test/session_form_extensions.rb` to accept pseudo-`name`s.